Core.lua

```diff
@@ -1192,30 +1177,689 @@ function Core:AppendTeam(teamData)
+					filteredMembers = backupGroupMembers
+				elseif specRole == "All" and teamGroup == "Trial" then
+					filteredMembers = trialGroupMembers
+				end
+
+				if specRole == "Tank" then
+					if teamGroup == "Main" then
+						filteredMembers = mainGroupTanks
+					elseif teamGroup == "Alt" then
+						filteredMembers = altGroupTanks
+					elseif teamGroup == "Backup" then
+						filteredMembers = backupGroupTanks
+					elseif teamGroup == "Trial" then
+						filteredMembers = trialGroupTanks
+					end
+				elseif specRole == "Healer" then
+					if teamGroup == "Main" then
+						filteredMembers = mainGroupHealers
+					elseif teamGroup == "Alt" then
+						filteredMembers = altGroupHealers
+					elseif teamGroup == "Backup" then
+						filteredMembers = backupGroupHealers
+					elseif teamGroup == "Trial" then
+						filteredMembers = trialGroupHealers
+					end
+				elseif specRole == "DPS" then
+					if teamGroup == "Main" then
+						filteredMembers = mainGroupDPS
+					elseif teamGroup == "Alt" then
+						filteredMembers = altGroupDPS
+					elseif teamGroup == "Backup" then
+						filteredMembers = backupGroupDPS
+					elseif teamGroup == "Trial" then
+						filteredMembers = trialGroupDPS
+					end
+				end
+			end
+
+			-- a local variable to help us render empty states
+			local totalTeamMembers = #filteredMembers
+
+			local function checkForEmptyState()
+				-- Render an empty state if no members are found.
+				if totalTeamMembers == 0 then
+					local noMembersLabel = GOW.GUI:Create("Label")
+					noMembersLabel:SetText("No members found.")
+					noMembersLabel:SetFullWidth(true)
+					if GoWTeamMemberContainer then
+						GoWTeamMemberContainer:AddChild(noMembersLabel)
+					end
+				end
+			end
+
+			checkForEmptyState()
+
+			-- Render each filtered member.
+			if filteredMembers then
+				-- Sort the members by online status.
+				table.sort(filteredMembers, function(a, b)
+					local function isOnline(member)
+						local num = GetNumGuildMembers()
+						for i = 1, num do
+							local name, _, _, _, _, _, _, _, online = GetGuildRosterInfo(i)
+							local baseName = name and name:match("^(.-)%-") or name
+							if baseName == member.name then
+								return online
+							end
+						end
+						return false
+					end
+					return isOnline(a) and not isOnline(b)
+				end)
+
+				-- attempt to find the member in the guild roster
+				for _, member in ipairs(filteredMembers) do
+					local isConnected = false
+					local numGuildMembers = GetNumGuildMembers()
+					local guildRankName = nil
+					local isInGuildOrCommunity = false
+
+					for i = 1, numGuildMembers do
+						local name, rankName, _, _, _, _, _, _, online = GetGuildRosterInfo(i)
+						-- In guild roster names may include realm (e.g. "Player-Realm"), so compare only the base names
+						local baseName = name and name:match("^(.-)%-") or name
+						if baseName == (member.name .. "-" .. member.realmNormalized) or baseName == member.name and online then
```

Comment:
This section can be simplified you can remove online from the if and set isConnected